src-java/rule-manager/rule-manager-implementation/src/main/java/org/openkilda/rulemanager/RuleManagerImpl.java

```diff
@@ -121,23 +123,21 @@ public RuleManagerImpl(RuleManagerConfig config) {
     @Override
     public List<SpeakerCommandData> buildRulesForSwitch(SwitchId switchId, DataAdapter adapter) {
         Switch sw = adapter.getSwitch(switchId);
-        SwitchProperties switchProperties = adapter.getSwitchProperties(switchId);
-
-        List<SpeakerCommandData> result = buildServiceRules(sw, switchProperties);
+        List<SpeakerCommandData> result = buildServiceRules(sw, adapter);
 
         result.addAll(buildFlowRulesForSwitch(switchId, adapter));
 
         return postProcessCommands(result);
     }
 
-    private List<SpeakerCommandData> buildServiceRules(Switch sw, SwitchProperties switchProperties) {
-        return getServiceRuleGenerators(switchProperties).stream()
+    private List<SpeakerCommandData> buildServiceRules(Switch sw, DataAdapter adapter) {
+        return getServiceRuleGenerators(sw.getSwitchId(), adapter).stream()
                 .flatMap(g -> g.generateCommands(sw).stream())
                 .collect(toList());
     }
 
     @VisibleForTesting
-    List<RuleGenerator> getServiceRuleGenerators(SwitchProperties switchProperties) {
+    List<RuleGenerator> getServiceRuleGenerators(SwitchId switchId, DataAdapter adapter) {
```

Comment:
I changed signature here to do not load ISL from DB if we don't need them. (when RTT is disabled)